_pytest/compat.py

```diff
@@ -237,5 +237,7 @@ def safe_str(v):
         try:
             return str(v)
         except UnicodeError:
+            if not isinstance(v, unicode):
+                v = unicode(v)
             errors = 'replace'
             return v.encode('ascii', errors)
```

Comment:
You got a point, I think it is safe to convert to a `utf-8` byte-stream, because it is `ascii` compatible anyway.

Updated.